### DISCUSSION TITLE
Pythia Cookbooks

### DISCUSSION THREAD

**Discussion by mgrover1** [ProjectPythia/projectpythia.github.io#170] [Ideas]
The idea of a "Pythia Cookbook" was brought up during the EWG meeting today. We thought it would be helpful to start a discussion on this here!

## What's a "Cookbook"?
A cookbook would be distinctly different from the Foundations content, such that it covers both:
* More advanced topics
* More **specific** use cases
* Enable an easier way for contributors to "plug their content in"

## What Are Examples of these Cookbooks?
It is easier to think of examples of what a cookbook topics:
* "Python Package Cookbook"
  * Designing tests for your code, continuous integration
  * Getting started with packaging your code and releasing on pypi + conda forge
  * How to get started with documentation
* "Station Data Cookbook"
  * Covering working with Pandas, NumPy for calculations, possibly overlaying other types of data
  * This book would refer to the Foundations content as "prerequisites"
* "Satellite Imagery Cookbook"
  * Covering working with satellite data, such as GOES-16 or LANDSAT
  * Detail working with netCDF Cloud Optimized Geo-TIFFs (COGs)
  * Perhaps the MetPy declarative interface for working with this type of data, or interactive plotting libraries
* "Cloud Model (or CESM) Cookbook"
  * How to work with model data, more advanced calculations using Xarray
  * Leverage Dask, other parallel or distributed libraries to be able to compute at scale

The main idea of these cookbooks is that they would reference the foundations material where possible, then dig into more advanced topics that help the learner understand the nuances of working with more use-specific libraries. This will enable us to dig into more advanced topics, and open the opportunity for others to get involved doesn't quite fit into the Foundations book.

**Comment by brian-rose** [comment]
Thanks @mgrover1 for articulating this so clearly. It seems to me that this "Cookbook" is the missing third leg of the Pythia content that was envisioned in the early days of the project. 

We started developing the Foundations book not just to help onboard new users but also to supply the "foundations" for a collection of advanced and/or domain-specific content. I suggest that a defining feature of the "Cookbook" that might distinguish it from other (non-Pythia) content collections could be an explicit and pervasive reliance on links to the Foundations content for necessary background. That was, after all, why we invested so much time in creating Foundations as a first step. 

Heavy reliance on Foundations will surely help drive further improvements and good maintenance of Foundations too, so there's a nice feedback loop there.

**Comment by clyne** [comment]
I also agree that Max's comments get to the topic of "what comes after foundations". Perhaps it is time to assess what we need to do to "finish" the foundations work, so we can turn our attention more to advanced topics. (of course with enough hands this work could proceed in parallel).  Perhaps this is also a good time to have some of the infrastructure group start thinking about what is needed to support "cookbooks"
